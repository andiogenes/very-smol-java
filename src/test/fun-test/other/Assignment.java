class Main {
    class Nested {
        int bar;
        int baz;
    }

    int bar() {
        return 10;
    }

    int x = bar();

    int foo() {
        return bar() * bar();
    }

    int y = foo() * bar();

    void main() {
        int z = x = y = 45;

        println(); println();

        Main.Nested.bar = Main.Nested.baz = x = y = z = foo() * bar();

        {
            int y = x = z = 45;
        }
        x = x;
        y = y;
        z = z;
    }
}