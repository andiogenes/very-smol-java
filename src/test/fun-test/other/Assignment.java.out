bar : INT = 0 : INT
baz : INT = 0 : INT
x : INT = 10 : INT
y : INT = 1000 : INT
y : INT = 45 : INT
x : INT = 45 : INT
z : INT = 45 : INT


z : INT = 1000 : INT
y : INT = 1000 : INT
x : INT = 1000 : INT
Main.Nested.baz : INT = 1000 : INT
Main.Nested.bar : INT = 1000 : INT
z : INT = 45 : INT
x : INT = 45 : INT
y : INT = 45 : INT
x : INT = 45 : INT
y : INT = 1000 : INT
z : INT = 45 : INT
